app/models/host.rb

```diff
@@ -87,6 +87,8 @@ class Host < ApplicationRecord
 
   has_many                  :host_service_groups, :dependent => :destroy
 
+  has_many                  :cloud_services, :dependent => :nullify
```

Comment:
D'oh, wish I'd seen this earlier.  This really belongs in `ManageIQ::Providers::Openstack::InfraManager::Host`.

....

But, I think what I'd rather do is start talking about creating a new type of Infrastructure Manager layer for Cloud Infrastructure.  Then, we could generalize some of these things that talk about "hosts that have services for clouds" and remove some of that logic from the underlying classes that were really meant for Virtual Infrastructure.

Longer term goal, so no change here ... just me talking to empty space.